<a href="https://colab.research.google.com/github/YYM89/MSSP607/blob/main/Par12.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [2]:
import pandas as pd
import scipy.stats as stats

# Load the dataframe
df = pd.read_csv('/content/drive/MyDrive/MSSP607/Data/Student Performance Data.csv')
print(df.head())
print(df.info())

   gender race/ethnicity parental level of education         lunch  \
0  female        group B           bachelor's degree      standard   
1  female        group C                some college      standard   
2  female        group B             master's degree      standard   
3    male        group A          associate's degree  free/reduced   
4    male        group C                some college      standard   

   math score  reading score  writing score  total score  \
0          72             72             74          218   
1          69             90             88          247   
2          90             95             93          278   
3          47             57             44          148   
4          76             78             75          229   

  test preparation course admission prospect  
0                    none             medium  
1               completed               high  
2                    none               high  
3                    none     

In [3]:
# Question 2 Analysis
# Group 1: Males who completed the preparation exam
group1_mask = (df['gender'] == 'male') & (df['test preparation course'] == 'completed')
group1 = df[group1_mask]

# Group 2: Others
group2 = df[~group1_mask]

# Perform t-tests for scores
scores = ['math score', 'reading score', 'writing score', 'total score']
results_q2 = {}

for score in scores:
    t_stat, p_val = stats.ttest_ind(group1[score], group2[score], equal_var=False)
    results_q2[score] = {'t_statistic': t_stat, 'p_value': p_val, 'mean_g1': group1[score].mean(), 'mean_g2': group2[score].mean()}

print("Results for Q2 (Males with Prep vs Others):")
print(pd.DataFrame(results_q2).T)

# Question 3 Analysis
# Comparison A: Gender (Male vs Female)
male = df[df['gender'] == 'male']
female = df[df['gender'] == 'female']

# Comparison B: Prep Course (Completed vs None)
prep_completed = df[df['test preparation course'] == 'completed']
prep_none = df[df['test preparation course'] == 'none']

# Comparison C: Lunch (Standard vs Free/Reduced)
lunch_std = df[df['lunch'] == 'standard']
lunch_free = df[df['lunch'] == 'free/reduced']

results_q3 = {}

# Helper function for t-test
def run_ttest(g1, g2, label_g1, label_g2, comparison_name):
    res = {}
    for score in scores:
        t_stat, p_val = stats.ttest_ind(g1[score], g2[score], equal_var=False)
        res[score] = {'t_stat': t_stat, 'p_val': p_val, f'mean_{label_g1}': g1[score].mean(), f'mean_{label_g2}': g2[score].mean()}
    return pd.DataFrame(res).T

df_gender = run_ttest(male, female, 'male', 'female', 'Gender')
df_prep = run_ttest(prep_completed, prep_none, 'completed', 'none', 'Prep Course')
df_lunch = run_ttest(lunch_std, lunch_free, 'standard', 'free/reduced', 'Lunch')

print("\nResults for Q3 - Gender:")
print(df_gender)
print("\nResults for Q3 - Prep Course:")
print(df_prep)
print("\nResults for Q3 - Lunch:")
print(df_lunch)

Results for Q2 (Males with Prep vs Others):
               t_statistic       p_value     mean_g1     mean_g2
math score        6.314235  1.161652e-09   72.339080   64.772397
reading score     1.092121  2.757703e-01   70.212644   68.949153
writing score     1.872093  6.220530e-02   69.793103   67.687651
total score       3.257452  1.269590e-03  212.344828  201.409201

Results for Q3 - Gender:
                 t_stat         p_val   mean_male  mean_female
math score     5.398001  8.420838e-08   68.728216    63.633205
reading score -7.968357  4.376297e-15   65.473029    72.608108
writing score -9.997719  1.711809e-22   63.311203    72.467181
total score   -4.178886  3.186198e-05  197.512448   208.708494

Results for Q3 - Prep Course:
                  t_stat         p_val  mean_completed   mean_none
math score      5.786983  1.042562e-08       69.695531   64.077882
reading score   8.004132  4.388808e-15       73.893855   66.534268
writing score  10.752510  2.662740e-25       74.418994   6

Yes, there are statistical differences, particularly in math and the total score, but the results vary by subject.

We compared Group 1 (Males who completed the test preparation course) against Group 2 (All other students).
-Math: Males who prepared scored significantly higher than the rest of the population (72.3 vs 64.8).
-Reading & Writing: There was no statistically significant difference in Reading, and the difference in Writing was marginal. This is likely because the "Others" group includes females, who generally perform better in reading and writing, balancing out the advantage that the males got from the preparation course.

Overall: The prepared males still had a significantly higher Total Score (212.3) compared to the rest of the students (201.4).

Are there any other differences that are of interest?

Yes, we performed additional t-tests on other groupings and found highly significant differences across several categories:

A. Gender (Male vs. Female)There is a distinct performance split based on subject.
-Math: Males score higher (68.7 vs 63.6, p < 0.001).
-Reading & Writing: Females score significantly higher (e.g., Writing: 72.5 vs 63.3, p < 0.001).Total Score: Overall, Females have a higher average total score (208.7 vs 197.5).

B. Test Preparation (Completed vs. None)
-The preparation course is associated with higher scores across all subjects.Students who completed the course scored significantly higher in Math (+5.6 points), Reading (+7.4 points), and Writing (+9.9 points) compared to those who did not.
-Total Score Difference: The prepared group averaged 218.0 compared to 195.1 for the unprepared group (p < 0.001).

C. Lunch Type (Standard vs. Free/Reduced)This variable, often a proxy for socioeconomic status, showed the strongest disparities.
-Students with Standard Lunch scored significantly higher in all three subjects.
-Total Score Difference: Standard lunch students averaged 212.5, while students with free/reduced lunch averaged 186.6, a massive difference of nearly 26 points (p < 0.001).